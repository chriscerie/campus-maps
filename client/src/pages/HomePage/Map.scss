.map-container {
  height: 92vh;
}
