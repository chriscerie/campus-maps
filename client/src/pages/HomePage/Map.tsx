import { useRef, useEffect } from 'react';
import mapboxgl from 'mapbox-gl';
import './Map.scss';

mapboxgl.accessToken =
  'pk.eyJ1IjoiY2hyaXNjZXJpZSIsImEiOiJja3VvcXBiaGExcG5vMnFtYjhnc3gxcGprIn0.eX9g2ClfVBqYEvecwIPLYw';

function Map() {
  const mapContainerRef = useRef<HTMLDivElement>(null);

  // Initialize map
  useEffect(() => {
    if (!mapContainerRef.current) {
      return;
    }

    const map = new mapboxgl.Map({
      container: mapContainerRef.current,
      // See style options here: https://docs.mapbox.com/api/maps/#styles
      style: 'mapbox://styles/mapbox/streets-v11',
      center: [-104.9876, 39.7405],
      zoom: 12.5,
    });

    // Navigation control (the +/- zoom buttons)
    map.addControl(new mapboxgl.NavigationControl(), 'bottom-right');

    return () => map.remove();
  }, []);

  return <div className="map-container" ref={mapContainerRef} />;
}

export default Map;
