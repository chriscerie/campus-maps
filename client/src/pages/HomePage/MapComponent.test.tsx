import ShallowRenderer from 'react-test-renderer/shallow';
import Map from './MapComponent';

describe('MapComponent', () => {
  it('should render successfully', () => {
    const renderer = ShallowRenderer.createRenderer();
    const tree = renderer.render(<Map />);
    expect(tree).toMatchSnapshot();
  });
});
