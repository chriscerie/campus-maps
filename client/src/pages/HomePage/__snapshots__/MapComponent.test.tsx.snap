// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapComponent should render successfully 1`] = `
<div
  className="map-container"
  style={
    Object {
      "height": 0,
      "visibility": "hidden",
    }
  }
/>
`;
